src/smo/xform/beamers.ts

```diff
@@ -149,40 +149,55 @@ export class SmoBeamer {
     }
 
     // beam tuplets
-    if (note.isTuplet) {
-      const tuplet = this.measure.getTupletForNote(note);
-      // The underlying notes must have been deleted.
-      if (!tuplet) {
-        return;
-      }
-      const tupletIndex = tuplet.getIndexOfNote(note);
-      const ult = tuplet.notes[tuplet.notes.length - 1];
-      const first = tuplet.notes[0];
+    // if (note.isTuplet) {
```

Comment:
Did we want to do something here?  There is no way currently to manage different beam levels in the Smoosic UI, e.g.  It works OK if it happens 'naturally', like eight + 2 16th, but there is no way to control where the split happens.  It would be ideal if a triplet with a nested triplet had a single beam across the duration (I think, I'll actually have to check what is the best practice).

![image](https://github.com/user-attachments/assets/2ae367ca-f052-477e-8e7f-2758e489dcbe)

Reply:
I think each level has its boundaries, and beam is never crossing them. I might be wrong though.
Regarding the beamer implementation, I changed it just enough so that it does not throw errors anymore.
Commented-out code is here for convenience when comparing the old version to the new one.
As you said, this part needs some work...